platform_setup.sh

```diff
@@ -1,62 +1,60 @@
-# ---------------------------------------------------------------------------- #
-# Valiant Platform application setup script.
-# ---------------------------------------------------------------------------- #
-
-# Install packages and services
-# https://www.notion.so/valiant/1-Install-packages-and-services-8a7e7d1dcaa74d829256ca4b4772fae2
-
-# Install XCode CLI tools - We require this for GIT
-sudo xcode-select --install
-
-# Install Homebrew - Package manager for macOS https://brew.sh/
-echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zshrc
-/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-
-# Install Heroku CLI & Heroku Multiple Accounts Plugin
-brew install heroku/brew/heroku
+#!/usr/bin/env bash
+
+function append_config() {
+  if [ ! -f $1 ]
+  then
+    return
+  fi
+  if grep -q "$2" "$1"
+  then
+    return
+  fi
+  echo "Appending to $1: $2"
+  echo >> "$1"
+  echo "# Added by platform_setup.sh" >> "$1"
+  echo "$2" >> "$1"
+}
+
+function append_file() {
+  append_config "$HOME/.zshrc" "$1"
+  append_config "$HOME/.bash_profile" "$1"
+}
```

Comment:
^ these functions append to `.zshrc` and/or `.bash_profile` only if they exist and do not already contain the line; the append is preceded by a comment so people know where this stuff came from